### PR TITLE
Added node template support for Opennebula

### PR DESCRIPTION
**Pull request by tibuntu**
This PR is a follow-up of the discussion in #105. 

Similar as in #216 the pre-condition is an installed Opennebula driver (See example below).

```
resource "rancher2_node_driver" "opennebula" {
    active = true
    builtin = false
    name = "opennebula"
    url = "https://github.com/OpenNebula/docker-machine-opennebula/releases/download/release-0.2.0/docker-machine-driver-opennebula.tgz"
}
```

This is a minimal example how to use the node template in Terraform:
```
resource "rancher2_node_template" "opennebula" {
  depends_on = [rancher2_node_driver.opennebula]
  name = "opennebula node template"
  description = "Some meaningful description"
  opennebula_config {
      user  = "<Opennebula user here>"
      password = "<Opennebula password here>"
      templateid = "<Some template id here>"
      xmlrpcurl = "<URL of the Opennebula XML RPC API>"
  }
}
```

See the official Opennebula docker machine documentation for all possible configs:
https://github.com/OpenNebula/docker-machine-opennebula#available-driver-options

### CODE REVIEW

File: rancher2/schema_node_template.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ type NodeTemplate struct {
 	DigitaloceanConfig  *digitaloceanConfig  `json:"digitaloceanConfig,omitempty" yaml:"digitaloceanConfig,omitempty"`
 	OpenstackConfig     *openstackConfig     `json:"openstackConfig,omitempty" yaml:"openstackConfig,omitempty"`
 	VmwarevsphereConfig *vmwarevsphereConfig `json:"vmwarevsphereConfig,omitempty" yaml:"vmwarevsphereConfig,omitempty"`
+	OpennebulaConfig    *opennebulaConfig    `json:"opennebula,omitempty" yaml:"opennebula,omitempty"`
```
Comment (rawmind0) [inline rancher2/schema_node_template.go]: Json and yaml definitions should be `opennebulaConfig`

```
-	OpennebulaConfig    *opennebulaConfig    `json:"opennebula,omitempty" yaml:"opennebula,omitempty"`
+	OpennebulaConfig    *opennebulaConfig    `json:"opennebulaConfig,omitempty" yaml:"opennebulaConfig,omitempty"`
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@tibuntu ,  thanks for the PR. This node template is not working due to the id for user created node_driver is missed. Have you tested it??

Following your instructions, getting error Error: `[ERROR] Getting Node Driver opennebula: Bad response statusCode [404]. Status [404 Not Found]. Body: [baseType=error, code=NotFound, message=nodedrivers.management.cattle.io "opennebula" not found] from [https://XX.XX.XX.XX/v3/nodedrivers/hetzner]`

I've submitted a PR to address this issue, adding driver_id argument to rancher2_node_template resource, #240 . Once it's merged, could you please rebase and test your PR with this tf file??

```
resource "rancher2_node_driver" "opennebula" {
    active = true
    builtin = false
    name = "opennebula"
    url = "https://github.com/OpenNebula/docker-machine-opennebula/releases/download/release-0.2.0/docker-machine-driver-opennebula.tgz"
}

resource "rancher2_node_template" "opennebula" {
  depends_on = [rancher2_node_driver.opennebula]
  name = "opennebula node template"
  description = "Some meaningful description"
  driver_id = rancher2_node_driver.opennebula.id
  opennebula_config {
      user  = "<Opennebula user here>"
      password = "<Opennebula password here>"
      templateid = "<Some template id here>"
      xmlrpcurl = "<URL of the Opennebula XML RPC API>"
  }
}
```

Also, node_template documentation and tests should be updated and included in the PR. For adding test, please take a look on rancher2/resource_rancher2_node_template_test.go and add another case for using your node_driver. You shouldn't provide token or any sensible data, take a look to other node_template tests.

**Comment by rawmind0** [comment]
This PR requires #240

File: rancher2/schema_node_template_opennebula.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+			Default:  "docker",
+		},
+		"templatename": &schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"vcpu": &schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
+		},
```
Comment (rawmind0) [inline rancher2/schema_node_template_opennebula.go:116]: Terraform argument names should use underscore case.

File: rancher2/schema_node_template_opennebula.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	cpu          string `json:"cpu,omitempty" yaml:"cpu,omitempty"`
+	disablevnc   bool   `json:"disableVnc,omitempty" yaml:"disableVnc,omitempty"`
+	imageid      string `json:"imageId,omitempty" yaml:"imageId,omitempty"`
+	imageowner   string `json:"imageOwner,omitempty" yaml:"imageOwner,omitempty"`
+	networkid    string `json:"networkId,omitempty" yaml:"networkId,omitempty"`
+	networkowner string `json:"networkOwner,omitempty" yaml:"networkOwner,omitempty"`
+	sshuser      string `json:"sshUser,omitempty" yaml:"sshUser,omitempty"`
+	templatename string `json:"templateName,omitempty" yaml:"templateName,omitempty"`
+	vcpu         string `json:"vcpu,omitempty" yaml:"vcpu,omitempty"`
+}
```
Comment (rawmind0) [inline rancher2/schema_node_template_opennebula.go:33]: Every field name on `opennebulaConfig` should start with a capital letter.

File: rancher2/structure_node_template_opennebula.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	obj["xmlrpcurl"] = in.xmlrpcurl
+	obj["cpu"] = in.cpu
+	obj["disableVnc"] = in.disablevnc
+	obj["imageId"] = in.imageid
+	obj["imageOwner"] = in.imageowner
+	obj["networkId"] = in.networkid
+	obj["networkOwner"] = in.networkowner
+	obj["sshUser"] = in.sshuser
+	obj["templateName"] = in.templatename
+	obj["vcpu"] = in.vcpu
```
Comment (rawmind0) [inline rancher2/structure_node_template_opennebula.go]: Some assignments are not correct, please review them.

File: rancher2/structure_node_template_opennebula.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	if v, ok := in["sshUser"].(string); ok && len(v) > 0 {
+		obj.sshuser = v
+	}
+	if v, ok := in["templateName"].(string); ok && len(v) > 0 {
+		obj.templatename = v
+	}
+	if v, ok := in["vcpu"].(string); ok && len(v) > 0 {
+		obj.vcpu = v
+	}
```
Comment (rawmind0) [inline rancher2/structure_node_template_opennebula.go:99]: Some assignments are not correct, please review them.

**Review by rawmind0 — CHANGES_REQUESTED** [review]
Please, take a look to the requested changes.

Also, node_template documentation and tests should be updated and included in the PR. For adding test, please take a look on rancher2/resource_rancher2_node_template_test.go and add another case for using your node_driver. You shouldn't provide token or any sensible data, take a look to other node_template tests.

File: rancher2/resource_rancher2_node_template_test.go [hunk 6]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
...
+  name = "foo"
+  description = "Terraform node driver opennbula acceptance test"
+  opennebula_config {
+	user = "apiuser"
+	password =  "password123"
+	ssh_user = "rancher"
+	template_name = "template-YYYYYYYY"
+	xml_rpc_url = "http://XXXXXXXX/RPC2"
+  }
+}
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: Here would be more useful if you modify some data on `opennebula_config` argument at `testAccRancher2NodeTemplateUpdateConfigOpennebula` to check that update is working properly.

File: rancher2/resource_rancher2_node_template_test.go [hunk 7]
```diff
@@ -540,6 +579,72 @@ func TestAccRancher2NodeTemplate_disappears_Vsphere(t *testing.T) {
...
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccRancher2NodeTemplateConfigOpennebula,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRancher2NodeTemplateExists(name, nodeTemplate),
+					resource.TestCheckResourceAttr(name, "name", "foo"),
+					resource.TestCheckResourceAttr(name, "description", "Terraform node driver opennebula acceptance test"),
+					resource.TestCheckResourceAttr(name, "driver", opennebulaConfigDriver),
+					resource.TestCheckResourceAttr(name, "opennebula_config.0.template_name", "template-YYYYYYYY"),
+					resource.TestCheckResourceAttr(name, "opennebula_config.0.ssh_user", "user-XXXXXXXX"),
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: This is not working due to `ssh_user` defined at `testAccRancher2NodeTemplateConfigOpennebula` is `rancher`

File: rancher2/resource_rancher2_node_template_test.go [hunk 8]
```diff
@@ -540,6 +579,72 @@ func TestAccRancher2NodeTemplate_disappears_Vsphere(t *testing.T) {
...
+			},
+			resource.TestStep{
+				Config: testAccRancher2NodeTemplateUpdateConfigOpennebula,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRancher2NodeTemplateExists(name, nodeTemplate),
+					resource.TestCheckResourceAttr(name, "name", "foo2"),
+					resource.TestCheckResourceAttr(name, "description", "Terraform node driver opennebula acceptance test - updated"),
+					resource.TestCheckResourceAttr(name, "driver", opennebulaConfigDriver),
+					resource.TestCheckResourceAttr(name, "opennebula_config.0.template_name", "template-YYYYYYYY"),
+					resource.TestCheckResourceAttr(name, "opennebula_config.0.ssh_user", "user-XXXXXXXX"),
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: This is not working due to `ssh_user` defined at `testAccRancher2NodeTemplateUpdateConfigOpennebula ` is `rancher`

File: rancher2/resource_rancher2_node_template_test.go [hunk 9]
```diff
@@ -540,6 +579,72 @@ func TestAccRancher2NodeTemplate_disappears_Vsphere(t *testing.T) {
...
+			},
+			resource.TestStep{
+				Config: testAccRancher2NodeTemplateRecreateConfigOpennebula,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRancher2NodeTemplateExists(name, nodeTemplate),
+					resource.TestCheckResourceAttr(name, "name", "foo"),
+					resource.TestCheckResourceAttr(name, "description", "Terraform node driver opennebula acceptance test"),
+					resource.TestCheckResourceAttr(name, "driver", opennebulaConfigDriver),
+					resource.TestCheckResourceAttr(name, "opennebula_config.0.template_name", "template-YYYYYYYY"),
+					resource.TestCheckResourceAttr(name, "opennebula_config.0.ssh_user", "user-XXXXXXXX"),
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: This is not working due to `ssh_user` defined at `testAccRancher2NodeTemplateRecreateConfigOpennebula ` is `rancher`

File: rancher2/resource_rancher2_node_template_test.go [hunk 10]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
 	pool =  "pool-XXXXXXXX"
   }
 }
+`
+	testAccRancher2NodeTemplateConfigOpennebula = `
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: You need to add node driver definition, due to is not Rancher builtin

```
-	testAccRancher2NodeTemplateConfigOpennebula = `
+	testAccRancher2NodeTemplateConfigOpennebulaDriver = `
+resource "rancher2_node_driver" "opennebula" {
+    active = true
+    builtin = false
+    name = "opennebula"
+    url = "https://github.com/OpenNebula/docker-machine-opennebula/releases/download/release-0.2.0/docker-machine-driver-opennebula.tgz"
+}
+`
+
+	testAccRancher2NodeTemplateConfigOpennebula = testAccRancher2NodeTemplateConfigOpennebulaDriver + `
```

File: rancher2/resource_rancher2_node_template_test.go [hunk 11]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
...
+  opennebula_config {
+	user = "apiuser"
+	password =  "password123"
+	ssh_user = "rancher"
+	template_name = "template-YYYYYYYY"
+	xml_rpc_url = "http://XXXXXXXX/RPC2"
+  }
+}
+`
+	testAccRancher2NodeTemplateUpdateConfigOpennebula = `
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: You need to add node driver definition

```
-	testAccRancher2NodeTemplateUpdateConfigOpennebula = `
+	testAccRancher2NodeTemplateUpdateConfigOpennebula = testAccRancher2NodeTemplateConfigOpennebulaDriver + `
```

File: rancher2/resource_rancher2_node_template_test.go [hunk 12]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
...
+  opennebula_config {
+  	user = "apiuser"
+	password =  "password123"
+	ssh_user = "rancher"
+	template_name = "template-YYYYYYYY"
+	xml_rpc_url = "http://XXXXXXXX/RPC2"
+  }
+}
+`
+	testAccRancher2NodeTemplateRecreateConfigOpennebula = `
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: You need to add node driver definition

```
-	testAccRancher2NodeTemplateRecreateConfigOpennebula = `
+	testAccRancher2NodeTemplateRecreateConfigOpennebula = testAccRancher2NodeTemplateConfigOpennebulaDriver + `
```

File: rancher2/schema_node_template_opennebula.go [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"network_name": &schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"password": &schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
```
Comment (rawmind0) [inline rancher2/schema_node_template_opennebula.go:66]: On documentation, `password` is set as required. Also, this argument should be sensitive to hide it on terraform execution.

```
-			Optional: true,
+			Required: true,
+			Sensitive: true,
```

File: website/docs/r/nodeTemplate.html.markdown [hunk 14]
```diff
@@ -180,6 +181,32 @@ The following attributes are exported:
...
+* `disk_resize` - (Optional) Size of the disk for the VM in MB (string)
+* `disable_vnc` - (Optional) VNC is enabled by default. Disable it with this flag (bool)
+* `image_id` - (Required*) Image ID to use as the VM OS (string)
+* `image_name` - (Required*) Opennebula image to use as the VM OS (string)
+* `image_owner` - (Optional) Owner of the image to use as the VM OS (string)
+* `memory`- (Optional) Size of the memory for the VM in MB (string)
+* `network_id` - (Optional) Opennebula network ID to connect the machine to (string)
+* `network_name` - (Optional) Opennebula network to connect the machine to (string)
+* `network_owner` - (Optional) Opennebula user ID of the Network to connect the machine to (string)
+* `password` - (Required) Set the password for the XML-RPC API authentication (string)
```
Comment (rawmind0) [inline website/docs/r/nodeTemplate.html.markdown]: This should be sensitive

```
-* `password` - (Required) Set the password for the XML-RPC API authentication (string)
+* `password` - (Required/Sensitive) Set the password for the XML-RPC API authentication (string)
```

File: website/docs/r/nodeTemplate.html.markdown [hunk 15]
```diff
@@ -180,6 +181,32 @@ The following attributes are exported:
...
+
+#### Arguments
+
+* `b2d_size` - (Optional) Size of the Volatile disk in MB - only for b2d (string)
+* `cpu` - (Optional) CPU value for the VM (string)
+* `dev_prefix` - (Optional) Dev prefix to use for the images. E.g.: 'vd', 'sd', 'hd' (string)
+* `disk_resize` - (Optional) Size of the disk for the VM in MB (string)
+* `disable_vnc` - (Optional) VNC is enabled by default. Disable it with this flag (bool)
+* `image_id` - (Required*) Image ID to use as the VM OS (string)
+* `image_name` - (Required*) Opennebula image to use as the VM OS (string)
```
Comment (rawmind0) [inline website/docs/r/nodeTemplate.html.markdown]: Could you please specify confilcting arguments?? 

```
-* `image_id` - (Required*) Image ID to use as the VM OS (string)
-* `image_name` - (Required*) Opennebula image to use as the VM OS (string)
+* `image_id` - (Required*) Opennebula image ID to use as the VM OS. Conflicts with `image_name` (string)
+* `image_name` - (Required*) Opennebula image name to use as the VM OS. Conflicts with `image_id` (string)
```

File: website/docs/r/nodeTemplate.html.markdown [hunk 16]
```diff
@@ -180,6 +181,32 @@ The following attributes are exported:
...
+* `image_owner` - (Optional) Owner of the image to use as the VM OS (string)
+* `memory`- (Optional) Size of the memory for the VM in MB (string)
+* `network_id` - (Optional) Opennebula network ID to connect the machine to (string)
+* `network_name` - (Optional) Opennebula network to connect the machine to (string)
```
Comment (rawmind0) [inline website/docs/r/nodeTemplate.html.markdown]: Could you please specify confilcting arguments??

```
-* `network_id` - (Optional) Opennebula network ID to connect the machine to (string)
-* `network_name` - (Optional) Opennebula network to connect the machine to (string)
+* `network_id` - (Optional) Opennebula network ID to connect the machine to. Conflicts with `network_name` (string)
+* `network_name` - (Optional) Opennebula network name to connect the machine to. Conflicts with `network_id` (string)
```

File: website/docs/r/nodeTemplate.html.markdown [hunk 17]
```diff
@@ -180,6 +181,32 @@ The following attributes are exported:
...
+* `network_owner` - (Optional) Opennebula user ID of the Network to connect the machine to (string)
+* `password` - (Required) Set the password for the XML-RPC API authentication (string)
+* `ssh_user` - (Optional) Set the name of the SSH user. Defaults to docker (string)
+* `template_id` - (Required*) Opennebula template ID to use (string)
+* `template_name` - (Required*) Name of the Opennbula template to use (string)
```
Comment (rawmind0) [inline website/docs/r/nodeTemplate.html.markdown]: Could you please specify confilcting arguments??

```
-* `template_id` - (Required*) Opennebula template ID to use (string)
-* `template_name` - (Required*) Name of the Opennbula template to use (string)
+* `template_id` - (Required*) Opennebula template ID to use. Conflicts with `template_name` (string)
+* `template_name` - (Required*) Opennbula template name to use. Conflicts with `template_id` (string)
```

File: website/docs/r/nodeTemplate.html.markdown [hunk 18]
```diff
@@ -8,9 +8,9 @@ description: |-
 
 # rancher2\_node\_template
 
-Provides a Rancher v2 Node Template resource. This can be used to create Node Template for Rancher v2 and retrieve their information. 
+Provides a Rancher v2 Node Template resource. This can be used to create Node Template for Rancher v2 and retrieve their information.
 
-amazonec2, azure, digitalocean, openstack and vsphere drivers are supported for node templates.
+amazonec2, azure, digitalocean, openstack, vsphere and opennebula drivers are supported for node templates.
```
Comment (rawmind0) [inline website/docs/r/nodeTemplate.html.markdown]: Could you please set it in alphabetical order??

```
-amazonec2, azure, digitalocean, openstack, vsphere and opennebula drivers are supported for node templates.
+amazonec2, azure, digitalocean, opennebula, openstack and vsphere drivers are supported for node templates.
```

File: website/docs/r/nodeTemplate.html.markdown [hunk 19]
```diff
@@ -180,6 +181,32 @@ The following attributes are exported:
...
+* `network_name` - (Optional) Opennebula network to connect the machine to (string)
+* `network_owner` - (Optional) Opennebula user ID of the Network to connect the machine to (string)
+* `password` - (Required) Set the password for the XML-RPC API authentication (string)
+* `ssh_user` - (Optional) Set the name of the SSH user. Defaults to docker (string)
+* `template_id` - (Required*) Opennebula template ID to use (string)
+* `template_name` - (Required*) Name of the Opennbula template to use (string)
+* `user` - (Required) Set the user for the XML-RPC API authentication (string)
+* `vcpu` - (Optional) VCPUs for the VM (string)
+* `xml_rpc_url` - (Required) Set the url for the Opennebula XML-RPC API (string)
+
```
Comment (rawmind0) [inline website/docs/r/nodeTemplate.html.markdown:207]: Could you please set required arguments first on the list??

File: rancher2/resource_rancher2_node_template_test.go [hunk 20]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
 	pool =  "pool-XXXXXXXX"
   }
 }
+`
+	testAccRancher2NodeTemplateConfigOpennebula = `
+resource "rancher2_node_template" "foo" {
+  name = "foo"
+  description = "Terraform node driver opennebula acceptance test"
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: `driver_id` argument is needed

```
+  driver_id = rancher2_node_driver.opennebula.id
```

File: rancher2/resource_rancher2_node_template_test.go [hunk 21]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
...
+	ssh_user = "rancher"
+	template_name = "template-YYYYYYYY"
+	xml_rpc_url = "http://XXXXXXXX/RPC2"
+  }
+}
+`
+	testAccRancher2NodeTemplateUpdateConfigOpennebula = `
+resource "rancher2_node_template" "foo" {
+  name = "foo2"
+  description = "Terraform node driver opennebula acceptance test - updated"
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: `driver_id` argument is needed

```
+  driver_id = rancher2_node_driver.opennebula.id
```

File: rancher2/resource_rancher2_node_template_test.go [hunk 22]
```diff
@@ -204,6 +204,45 @@ resource "rancher2_node_template" "foo" {
...
+	ssh_user = "rancher"
+	template_name = "template-YYYYYYYY"
+	xml_rpc_url = "http://XXXXXXXX/RPC2"
+  }
+}
+`
+	testAccRancher2NodeTemplateRecreateConfigOpennebula = `
+resource "rancher2_node_template" "foo" {
+  name = "foo"
+  description = "Terraform node driver opennbula acceptance test"
```
Comment (rawmind0) [inline rancher2/resource_rancher2_node_template_test.go]: `driver_id` argument is needed

```
+  driver_id = rancher2_node_driver.opennebula.id
```

File: rancher2/schema_node_template_opennebula.go (hunk 2 above)
Comment (rawmind0) [reply rancher2/schema_node_template_opennebula.go:116]: Could you please set required arguments first on the list??

File: rancher2/schema_node_template_opennebula.go (hunk 3 above)
Comment (rawmind0) [reply rancher2/schema_node_template_opennebula.go:33]: Could you please set fields in alphabetical order??

File: rancher2/schema_node_template.go [hunk 23]
```diff
@@ -28,7 +29,7 @@ func nodeTemplateFields() map[string]*schema.Schema {
 			Type:          schema.TypeList,
 			MaxItems:      1,
 			Optional:      true,
-			ConflictsWith: []string{"azure_config", "digitalocean_config", "openstack_config", "vsphere_config"},
+			ConflictsWith: []string{"azure_config", "digitalocean_config", "openstack_config", "vsphere_config", "opennebula_config"},
```
Comment (rawmind0) [inline rancher2/schema_node_template.go]: Could you please set it in alphabetical order??

```
-			ConflictsWith: []string{"azure_config", "digitalocean_config", "openstack_config", "vsphere_config", "opennebula_config"},
+			ConflictsWith: []string{"azure_config", "digitalocean_config", "opennebula_config", "openstack_config", "vsphere_config"},
```

File: rancher2/schema_node_template.go [hunk 24]
```diff
@@ -47,7 +48,7 @@ func nodeTemplateFields() map[string]*schema.Schema {
 			Type:          schema.TypeList,
 			MaxItems:      1,
 			Optional:      true,
-			ConflictsWith: []string{"amazonec2_config", "digitalocean_config", "openstack_config", "vsphere_config"},
+			ConflictsWith: []string{"amazonec2_config", "digitalocean_config", "openstack_config", "vsphere_config", "opennebula_config"},
```
Comment (rawmind0) [inline rancher2/schema_node_template.go]: Could you please set it in alphabetical order??

```
-			ConflictsWith: []string{"amazonec2_config", "digitalocean_config", "openstack_config", "vsphere_config", "opennebula_config"},
+			ConflictsWith: []string{"amazonec2_config", "digitalocean_config", "opennebula_config", "openstack_config", "vsphere_config"},
```

File: rancher2/schema_node_template.go [hunk 25]
```diff
@@ -64,7 +65,7 @@ func nodeTemplateFields() map[string]*schema.Schema {
 			Type:          schema.TypeList,
 			MaxItems:      1,
 			Optional:      true,
-			ConflictsWith: []string{"amazonec2_config", "azure_config", "openstack_config", "vsphere_config"},
+			ConflictsWith: []string{"amazonec2_config", "azure_config", "openstack_config", "vsphere_config", "opennebula_config"},
```
Comment (rawmind0) [inline rancher2/schema_node_template.go]: Could you please set it in alphabetical order??

```
-			ConflictsWith: []string{"amazonec2_config", "azure_config", "openstack_config", "vsphere_config", "opennebula_config"},
+			ConflictsWith: []string{"amazonec2_config", "azure_config", "opennebula_config", "openstack_config", "vsphere_config"},
```

File: rancher2/schema_node_template.go [hunk 26]
```diff
@@ -117,7 +118,7 @@ func nodeTemplateFields() map[string]*schema.Schema {
 			Type:          schema.TypeList,
 			MaxItems:      1,
 			Optional:      true,
-			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "vsphere_config"},
+			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "vsphere_config", "opennebula_config"},
```
Comment (rawmind0) [inline rancher2/schema_node_template.go]: Could you please set it in alphabetical order??

```
-			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "vsphere_config", "opennebula_config"},
+			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "opennebula_config", "vsphere_config"},
```

File: rancher2/schema_node_template.go [hunk 27]
```diff
@@ -131,11 +132,20 @@ func nodeTemplateFields() map[string]*schema.Schema {
 			Type:          schema.TypeList,
 			MaxItems:      1,
 			Optional:      true,
-			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "openstack_config"},
+			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "openstack_config", "opennebula_config"},
```
Comment (rawmind0) [inline rancher2/schema_node_template.go]: Could you please set it in alphabetical order??

```
-			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "openstack_config", "opennebula_config"},
+			ConflictsWith: []string{"amazonec2_config", "azure_config", "digitalocean_config", "opennebula_config", "openstack_config"},
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@tibuntu  thanks for the update. Please take a look to the changes requested. 

Before commit, please run the acceptance test to check that are passing, just docker is required.
osx: `TESTACC_EXPOSE_HOST_PORTS=true make testacc` 
linux: `make testacc`

**Comment by tibuntu** [comment]
Thank you once more for all the feedback. This is literally the first time I am submitting a PR for an upstream project so apologizes for all the struggle!

**Review by rawmind0 — APPROVED** [review]
@tibuntu thanks for the update. Acceptance tests are passing just fine.

LGTM

**Pull request merged**